### PR TITLE
fix idempotency on planning building and building unit

### PR DESCRIPTION
**Pull request by emalfroy**

### CODE REVIEW

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/Building/PlanBuildingLambdaHandler.cs [hunk 1]
```diff
@@ -13,38 +13,32 @@ namespace BuildingRegistry.Api.BackOffice.Handlers.Lambda.Handlers.Building
...
-        }
+        { }
 
         protected override async Task<ETagResponse> InnerHandle(PlanBuildingLambdaRequest request, CancellationToken cancellationToken)
         {
-            var buildingPersistentLocalId = new BuildingPersistentLocalId(_persistentLocalIdGenerator.GenerateNextPersistentLocalId());
-            var cmd = request.ToCommand(buildingPersistentLocalId);
+            var cmd = request.ToCommand();
 
             await IdempotentCommandHandler.Dispatch(
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/Building/PlanBuildingLambdaHandler.cs:34]: I think we should add a try catch where we catch the idempotency exception.

File: src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/PlanBuildingUnitLambdaHandler.cs [hunk 2]
```diff
@@ -36,14 +34,11 @@ public sealed class PlanBuildingUnitLambdaHandler : BuildingUnitLambdaHandler<Pl
...
         }
 
         protected override async Task<ETagResponse> InnerHandle(PlanBuildingUnitLambdaRequest request, CancellationToken cancellationToken)
         {
-            var buildingUnitPersistentLocalId = new BuildingUnitPersistentLocalId(_persistentLocalIdGenerator.GenerateNextPersistentLocalId());
-
-            var cmd = request.ToCommand(buildingUnitPersistentLocalId);
+            var cmd = request.ToCommand();
 
             await using var transaction = await _backOfficeContext.Database.BeginTransactionAsync(cancellationToken);
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/PlanBuildingUnitLambdaHandler.cs:44]: Again a try catch for the idempotency exception.

File: src/BuildingRegistry.Api.BackOffice/Infrastructure/Modules/ApiModule.cs [hunk 3]
```diff
@@ -59,10 +62,19 @@ protected override void Load(ContainerBuilder builder)
...
+
+            builder
+                .RegisterType<PlanBuildingSqsRequestFactory>()
+                .AsSelf();
+
             builder
                 .RegisterModule(new BackOfficeModule(_configuration, _services, _loggerFactory))
                 .RegisterModule(new MediatRModule())
                 .RegisterModule(new AggregateSourceModule(_configuration))
+                .RegisterModule(new SequenceModule(_configuration, _services, _loggerFactory))
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice/Infrastructure/Modules/ApiModule.cs:77]: Are we missing the snapshot module?

File: src/BuildingRegistry.Api.BackOffice/Infrastructure/Modules/ApiModule.cs [hunk 4]
```diff
@@ -60,14 +62,25 @@ protected override void Load(ContainerBuilder builder)
...
                 .RegisterModule(new SqsHandlersModule(_configuration[SqsQueueUrlConfigKey]))
                 .RegisterModule(new TicketingModule(_configuration, _services))
                 .RegisterModule(new ConsumerAddressModule(_configuration, _services, _loggerFactory));
 
+            builder.RegisterSnapshotModule(_configuration);
```
Comment (jvandaal) [inline src/BuildingRegistry.Api.BackOffice/Infrastructure/Modules/ApiModule.cs:82]: RegisterSnapshotModule can go as it is registered in AggregateSourceModule

**Review by pgallik — APPROVED** [review]

**Pull request merged**